apim/templates/ui/ui-configmap.yaml

```diff
@@ -29,6 +29,9 @@ data:
       {{- if or .Values.gateway.enabled .Values.ui.title .Values.ui.portal }},
       "portal": {
         {{- if .Values.gateway.enabled }}
+        {{- if .Values.gateway.entrypoint }}
```

Comment:
I'm not sure why we would need this kind of change.

Maybe you can explain it by providing an example and, better, provide some documentation about this property and its usage ?

Reply:
This is due to the way `nginx.ingress.kubernetes.io/rewrite-target` now works (since v0.22.0). 
Before 0.22, this: `nginx.ingress.kubernetes.io/rewrite-target: "/"` means that `GET ingress/toto` forwards to `mysvc/toto`
From 0.22, this:  `nginx.ingress.kubernetes.io/rewrite-target: "/"` means that `GET ingress/toto` forwards to `mysvc/`

This is due to the warning explicited [here](https://kubernetes.github.io/ingress-nginx/examples/rewrite/#rewrite-target) ` In Version 0.22.0 and beyond, any substrings within the request URI that need to be passed to the rewritten path must explicitly be defined in a capture group.`

If your ingress controller is updated, using a matching group in ingress path (as for example `(.*)`) will break the entrypoint value, meaning that any link from the ui to the gateway for `mygreatapi/v1` will look like `https://myapi.ingress.com/gateway/(.*)mygreatapi/v1`

Using `ingress.path` might even be counter indicative for this purpose